lib/galaxy/datatypes/util/generic_util.py

```diff
@@ -16,3 +16,21 @@ def count_special_lines(word, filename, invert=False):
     except commands.CommandLineException:
         return 0
     return int(out)
+
+
+def call_pysam_index(self, file_name, index_name, index_flag=None, stderr=None):
+    """
+    The pysam.index call can block the GIL, which can pause all threads, including
+    the heartbeat thread. Therefore, start it as an external process.
+    """
+    if index_flag == '-b' or not index_flag:
+        # IOError: No such file or directory: '-b' if index_flag is set to -b (pysam 0.15.4)
+        cmd = ['python', '-c', f"import pysam; pysam.set_verbosity(0); pysam.index('{file_name}', '{index_name}')"]
+    else:
+        cmd = ['python', '-c',
+               f"import pysam; pysam.set_verbosity(0); pysam.index('{index_flag}', '{file_name}', '{index_name}')"]
+    if stderr:
+        with open(stderr, 'w') as stderr:
+            subprocess.check_call(cmd, stderr=stderr, shell=False)
```

Comment:
Can you use https://github.com/galaxyproject/galaxy/blob/ca442596c344742c2a7f64d8d0bedac41536c88b/lib/galaxy/util/commands.py#L88?

Reply:
Thanks for reviewing. The original code has a specific comment saying that stderr needs to be discarded: https://github.com/galaxyproject/galaxy/pull/11558/files#diff-5da1157c80efcdb0e8e2232b02ea730d8b7bf0bcb2fbe6afb84e2828eec92a3fL490
and `execute` doesn't seem to support stderr redirection?

Reply:
That's what `stderr=subprocess.PIPE` does

Reply:
Aah I see, the piped stderr is being ignored. Sure, seems fine, can do.